quinn-h3/src/server.rs

```diff
@@ -154,10 +153,26 @@ impl RecvRequest {
             state: RecvRequestState::Receiving(FrameDecoder::stream(recv), send),
         }
     }
+
+    fn build_request(headers: Header) -> Result<Request<()>, Error> {
+        let (method, uri, headers) = headers.into_request_parts()?;
+        let mut request = Request::builder();
+        request.method(method);
+        request.uri(uri);
+        request.version(http::version::Version::HTTP_3);
+        match request.headers_mut() {
+            Some(h) => *h = headers,
+            None => return Err(Error::peer("invalid header")),
```

Comment:
When can this error arise?

Reply:
The [doc](https://docs.rs/http/0.1.18/http/request/struct.Builder.html#method.headers_mut) isn't clear about that, so I preferred not to `panic!`, and didn't bother to write a more specific error.

I guess the error could be triggered by an invalid status code (which is still transferred in ascii).  These values are received from the client.

Reply:
Maybe I should use ok_or_else though.